### PR TITLE
Fix the mapping btw pyspark and numpy

### PR DESCRIPTION
**Pull request by voe09**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [x] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes # (issue).
Horovod maps pyspark string type to numpy uint8 type. It blocks users using string type tensor input. I think originally the reason why we map string type to numpy uint8 is because tensorflow does not support string label. But with the mapping, horovod KerasModel would fail in transform stage anyway. And keras itself does not support string label (https://stackoverflow.com/questions/55970851/tensorflow-model-fit-does-not-work-with-strings-as-labels). We don't need to use the incorrect mapping to fail the transform

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Comment by chongxiaoc** [comment]
@voe09 
Surprisingly this function is not used in horovod? Which example did you use to catch this?

```
(hvd-py3.7) chongxiaoc@chongxiaoc-C02DL7WPMD6R horovod % ag data_type_to_numpy
horovod/spark/common/util.py
127:def data_type_to_numpy(dtype):
```

@tgaddair Do you remember a reason to use `uint8` for string type?

**Comment by chongxiaoc** [comment]
@voe09  Just saw your description now I get it.

**Review by tgaddair — APPROVED** [review]

**Comment by chongxiaoc** [comment]
@voe09 plz sign off so we can merge this.
